src/ClientAppSpec.ts

```diff
@@ -139,6 +139,90 @@ export default describe('ClientApp', () => {
                 });
             });
         });
+        describe('gcHostOriginQueryParam and gcTargetEnvQueryParam config', () => {
+            it('should allow a user to pass valid query param names into the constructor', () => {
+                // External
+                let query = '?gcHostOrigin=https://apps.mypurecloud.com&gcTargetEnv=prod';
+                spyOn(ClientApp, '_getQueryString').and.callFake(() => query);
+                let myClientApp = new ClientApp({
+                    gcHostOriginQueryParam: 'gcHostOrigin',
+                    gcTargetEnvQueryParam: 'gcTargetEnv'
+                });
+                expect(myClientApp.pcEnvironment).toBe('mypurecloud.com');
+                // Localhost
+                query = '?gcHostOrigin=https://localhost:1337&gcTargetEnv=prod';
+                myClientApp = new ClientApp({
+                    gcHostOriginQueryParam: 'gcHostOrigin',
+                    gcTargetEnvQueryParam: 'gcTargetEnv'
+                });
+                expect(myClientApp.pcEnvironment).toBe('localhost');
```

Comment:
too bad we dont have access to the full `_pcEnv` prop to assert the value of pcAppOrigin.  But, i've never been much of a fan of exposing private members solely for the purpose of tests.

I dont see a great way of asserting it elsewhere either

Reply:
actually, you found a great solution by testing `lookupGcEnv` in the envSpec.  I rounded out a few tests so we could better assert the parsed values.  But, also think we should keep these here to test the constructor 👍